src/swarm/neo/client/mixins/AllNodesRequestCore.d

```diff
@@ -0,0 +1,648 @@
+        handle_status_code = instance of HandleStatusCode
+
+    Returns:
+        instance of AllNodesRequestInitialiser constructed with the provided
+        arguments
+
+*******************************************************************************/
+
+public AllNodesRequestInitialiser!(Request, FillPayload, HandleStatusCode)
+    createAllNodesRequestInitialiser ( Request, FillPayload,
+    HandleStatusCode )
+    ( RequestOnConn.EventDispatcherAllNodes conn,
+    Request.Context* context, FillPayload fill_payload,
+    HandleStatusCode handle_status_code )
+{
+    return
+        AllNodesRequestInitialiser!(Request, FillPayload, HandleStatusCode)
+        (conn, context, fill_payload, handle_status_code);
+}
+
+///
+unittest
+{
+    // Example request struct with RequestCore mixed in
+    struct ExampleRequest
+    {
+        mixin ExampleRequestCore!();
+    }
+
+    // (Partial) request implementation, to be instantiated from the request's
+    // handler function.
+    scope class ExampleRequestImpl
+    {
+        import swarm.neo.connection.RequestOnConnBase;
+
+        private RequestOnConn.EventDispatcherAllNodes conn;
+        private ExampleRequest.Context* context;
+
+        public this ( RequestOnConn.EventDispatcherAllNodes conn,
+            ExampleRequest.Context* context )
+        {
+            this.conn = conn;
+            this.context = context;
+        }
+
+        // Instantiates an AllNodesRequestInitialiser with dummy delegates
+        // passed as the various policies.
+        public void run ( )
+        {
+            auto initialiser = createAllNodesRequestInitialiser!(ExampleRequest)(
+                this.conn, this.context, &this.fillPayload,
+                &this.handleStatusCode);
+
+            // Pass initialiser to an AllNodesRequest...
+        }
+
+        // Dummy policies...
+        private void fillPayload (
+            RequestOnConnBase.EventDispatcher.Payload payload ) { }
+
+        private bool handleStatusCode ( ubyte status )
+        {
+            return true;
+        }
+    }
+}
+
+/*******************************************************************************
+
+    Data required by AllNodesRequestInitialiser, to be placed in the request's
+    shared working data (the field must be named all_nodes).
+
+*******************************************************************************/
+
+public struct AllNodesRequestSharedWorkingData
+{
+    import swarm.neo.client.NotifierTypes;
+
+    /***************************************************************************
+
+        The number of request-on-conns that are currently in the process of
+        initialising the request -- sending the initial payload and waiting for
+        a status code back from the node.
+
+        Used to decide when to send the started notification to the user.
+
+    ***************************************************************************/
+
+    public uint initialising;
```

Comment:
I wonder if we make all this field `private`, would that make compiler enforce that it's touched only by the other entities in this module. This would enforce that the user defined code doesn't change this field.